src/scalar/union.rs

```diff
@@ -0,0 +1,48 @@
+use std::sync::Arc;
+
+use crate::datatypes::DataType;
+
+use super::Scalar;
+
+/// A single entry of a [`crate::array::UnionArray`].
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnionScalar {
+    value: Option<Arc<dyn Scalar>>,
```

Comment:
```suggestion
    value: Arc<dyn Scalar>,
```

unions do not have a validity. Thus, the value of an index is always "Some".